src/set.c

```diff
@@ -561,36 +559,26 @@ static Obj FuncUNITE_SET(Obj self, Obj set1, Obj set2)
         e1 = ELM_PLIST( set1, i1 );
```

Comment:
My main 'worry' about this function is that `e1` and `e2` are both re-accessed on every iteration of this loop, even though 'typically' only one of `e1` and `e2` has changed. I didn't immediately come up with a concise way of only accessing when necessary. Perhaps there's basically no cost to doing `ELM_PLIST` anyway?

Reply:
There *is* some cost to it, although it's low. So I agree there are further improvements possible in this function! My main goal with this patch was to make the code easier to understand (I actually created this months ago, while looking at something else -- forgot what -- and struggled a bit verifying the copying loop in the end. The new code actually might be a (very small) de-optimization, but IMHO here clarity is more important -- we already had some subtle bugs in code related to computing unions...

Reply:
Sure. I agree that simplicity is a good thing to aim for (and you were clear that that was your goal).

Reply:
(I'm happy for the PR to be merged as-is.)